apis/conditions/conditions.go

```diff
@@ -23,13 +24,15 @@ const (
 	TypeProvisioning = "Provisioning"
 	// TypeCreating indicates that an instance is being created for the first time.
 	TypeCreating = "Creating"
+	// TypeProgressing indicates that an instance is being updated.
+	TypeProgressing = "Progressing"
 )
 
 // Ready creates a condition with TypeReady, ReasonReady and empty message.
-func Ready() metav1.Condition {
+func Ready(status metav1.ConditionStatus) metav1.Condition {
 	return metav1.Condition{
 		Type:               TypeReady,
-		Status:             metav1.ConditionTrue,
+		Status:             status,
 		LastTransitionTime: metav1.Now(),
 		Reason:             ReasonReady,
 	}
```

Comment:
Nitpick: I think it would make sense to have 2 dedicated helper functions that don't accept a parameter, like this
```
func Ready() ...
func NotReady() ...
```
Or maybe better yet, the builder has a function that can set the status to false if so desired, e.g.
```
conditions.Builder().With(conditions.Ready()).WithStatus(metav1.ConditionFalse).Build()
```

Reply:
Let's go with the first option, builder is a bit of an overkill here for now